src/models/SetPacking.jl

```diff
@@ -0,0 +1,57 @@
+"""
+$TYPEDEF
+
+The [set packing problem](https://queracomputing.github.io/GenericTensorNetworks.jl/dev/generated/SetPacking/), a generalization of independent set problem to hypergraphs.
+
+Positional arguments
+-------------------------------
+* `sets` is a vector of vectors, each set is associated with a weight specified in `weights`.
+
+Currently this type problem doesn't support weights.
+
+Examples
+-------------------------------
+Under Development
+"""
+struct SetPacking{ET} <: AbstractProblem
+    sets::Vector{Vector{ET}}
+    function SetPacking(sets::Vector{Vector{ET}} ) where {ET}
+        new{ET}(sets)
+    end
+end
+Base.:(==)(a::SetPacking, b::SetPacking) = ( a.sets == b.sets )
+
+# Variables Interface
+variables(c::SetPacking) = [1:length(c.sets)...]
+flavors(::Type{<:SetPacking}) = [0, 1]
+
+"""
+    evaluate(c::SetPacking, config)
+
+* First step: We check if `config` (a vector of boolean numbers as the mask of sets) is a set packing of `sets`;
+* Second step: If it is a set packing, we return - (size(set packing)); Otherwise, we return Inf.
+"""
+function evaluate(c::SetPacking, config)
+    @assert length(config) == num_variables(c)
+    if is_set_packing(c.sets, config)
+        return - count(x -> x == 1, config)
+    else
+        return Inf
```

Comment:
`Inf` has type Float64, this function is not type stable.

Reply:
Change "inf" to "length(config) + 1"